### PR TITLE
Added two optional keyword arguments to show_rdkit()

### PR DESCRIPTION
**Pull request by ptosco**
- added an optional `confId` keyword argument to `show_rdkit()`
- added an optional `fmt` keyword argument to `show_rdkit()`; if `fmt` is not provided, a simple heuristic is used to guess if the RDKit molecule should be converted to a SDF or a PDB block

### CODE REVIEW

File: nglview/show.py [hunk 1]
```diff
@@ -359,7 +367,18 @@ def show_rdkit(rdkit_mol, **kwargs):
     >>> view = nv.show_rdkit(m, parmed=True) # doctest: +SKIP
     '''
     from rdkit import Chem
-    fh = StringIO(Chem.MolToPDBBlock(rdkit_mol))
+    try:
```
Comment (hainm) [inline nglview/show.py]: you can remove the whole try/except via

```
confId = kwargs.pop("confId", -1) 
```

File: nglview/show.py [hunk 2]
```diff
@@ -359,7 +367,18 @@ def show_rdkit(rdkit_mol, **kwargs):
...
+        confId = kwargs.pop("confId")
+    except KeyError:
+        confId = -1
+    try:
+        fmt = kwargs.pop("fmt")
+    except KeyError:
+        has_residue_info = (rdkit_mol.GetNumAtoms() and
+            rdkit_mol.GetAtomWithIdx(0).GetPDBResidueInfo())
+        fmt = "pdb" if has_residue_info else "sdf"
+    mol_read_fn = Chem.MolToPDBBlock if fmt == "pdb" else Chem.MolToMolBlock
```
Comment (hainm) [inline nglview/show.py]: change `mol_read_fn` to `mol_reader` or just `reader`

File: nglview/show.py [hunk 3]
```diff
@@ -359,7 +367,18 @@ def show_rdkit(rdkit_mol, **kwargs):
...
     from rdkit import Chem
-    fh = StringIO(Chem.MolToPDBBlock(rdkit_mol))
+    try:
+        confId = kwargs.pop("confId")
+    except KeyError:
+        confId = -1
+    try:
+        fmt = kwargs.pop("fmt")
+    except KeyError:
+        has_residue_info = (rdkit_mol.GetNumAtoms() and
```
Comment (hainm) [inline nglview/show.py]: you can combine few lines

```
if rdkit_mol.GetNumAtoms() and rdkit_mol.GetAtomWithIdx(0).GetPDBResidueInfo():
    fmt = 'pdb
else:
    fmt = 'sdf'
```
```

File: nglview/show.py [hunk 4]
```diff
@@ -337,6 +337,14 @@ def show_rdkit(rdkit_mol, **kwargs):
     ----------
     rdkit_mol : rdkit.Chem.rdchem.Mol
     kwargs : additional keyword argument
```
Comment (hainm) [inline nglview/show.py]: Can you please add a test with new options to make sure the code is excised? 

https://github.com/arose/nglview/blob/4dd648de75f5f0e769651d725d2491b168bdec11/nglview/tests/test_widget.py#L584

**Comment by ptosco** [comment]
@hainm Thanks for your comments, I have updated my PR accordingly. Happy to modify it further if needed.

**Comment by hainm** [comment]
@ptosco I've made some changes, please merge my change to your branch. thanks.
LGTM

rdkit

**Comment by ptosco** [comment]
@hainm Thanks Hai, this is now done.

**Comment by hainm** [comment]
Thank you.

**Pull request merged**